src/esm_environment/esm_environment.py

```diff
@@ -154,7 +155,8 @@ def apply_model_changes(self, model, run_or_compile="runtime", modelconfig=None)
 
             # Perform the merging of the environment dictionaries
             if "environment_changes" in modelconfig:
-                modelconfig["environment_changes"].update(modelconfig[thesechanges])
+                #ipdb.set_trace()
+                esm_parser.dict_merge(modelconfig["environment_changes"], modelconfig[thesechanges])
```

Comment:
I somehow feel the mistake should be more than just this line. Would it worthwhile to write a test for this logic, so one doesn't need to fire up the whole machinery?